File: python/simple_pvr/pvr_initializer.py
from datetime import timedelta
import os
from time import sleep

from .master_import import HDHomeRun
from .master_import import RecordingManager
from .master_import import Scheduler

from .server import app

__hdhomerun = HDHomeRun()
recordings_path = app.config['RECORDINGS_PATH'] if 'RECORDINGS_PATH' in app.config else None
__recording_manager = RecordingManager(recordings_directory=recordings_path)
__scheduler = Scheduler()

def scheduler():
    return __scheduler

def recording_manager():
    return __recording_manager

def hdhomerun():
    return __hdhomerun

def setup():
    from .master_import import Channel

    __scheduler.setDaemon(True)
    __scheduler.start()

    if not Channel.query.all():
        __hdhomerun.scan_for_channels()

def sleep_forever():
    forever = timedelta(days=6000)
    sleep(forever)

File: ruby/lib/simple_pvr/pvr_initializer.rb
module SimplePvr
  class PvrInitializer
    def self.setup
      Model::DatabaseInitializer.setup
      setup_with_hdhomerun(HDHomeRun.new)
      @hdhomerun.scan_for_channels if Model::Channel.all.empty?
    end
    
    def self.setup_for_integration_test
      Model::DatabaseInitializer.prepare_for_test
      setup_with_hdhomerun(HDHomeRunFake.new)
    end
    
    def self.hdhomerun
      @hdhomerun
    end
    
    def self.recording_manager
      @recording_manager
    end
    
    def self.scheduler
      @scheduler
    end

    def self.rackup_file_path
      File.dirname(__FILE__) + '/server/config.ru'
    end
    
    def self.rack_maps_file
      File.read(File.dirname(__FILE__) + '/server/rack_maps.rb')
    end
    
    def self.sleep_forever
      forever = 6000.days
      sleep forever
    end
    
    private
    def self.setup_with_hdhomerun(hdhomerun)
      @hdhomerun = hdhomerun
      @recording_manager = RecordingManager.new
      @scheduler = Scheduler.new
      @scheduler.start
    end
  end
end